src/index.tsx

```diff
@@ -241,21 +248,26 @@ const NextTopLoader = ({
             newUrl.startsWith(scheme)
           );
 
-          const isAnchor: boolean = isAnchorOfCurrentUrl(currentUrl, newUrl);
           const notSameHost = !isSameHostName(window.location.href, anchor.href);
           if (notSameHost) {
             return;
           }
+
+          const isAnchorOrHashAnchor =
+            isAnchorOfCurrentUrl(currentUrl, newUrl) || isHashAnchor(window.location.href, anchor.href);
```

Comment:
`isAnchorOfCurrentUrl` and `isHashAnchor` are very similar function that do almost the exact same thing. I'm not sure if both are needed but I am keeping both for this PR to keep existing behaviour. Can likely be merged into one function since they are only used in one spot unless I am missing something.

Reply:
Yeah, both functions do the same thing. I will refactor both functions to one function later.